AstSemantics.md

```diff
@@ -111,6 +112,9 @@ Integer loads can specify a *storage size* which is smaller than the result type
 well as a signedness which determines whether the bytes are sign- or zero-
 extended into the result type.
 
+Each `load` and `store` specifies the [linear memory index](Modules.md#index-spaces)
+to operate on with an unsigned integer immediate.
+
```

Comment:
It seems very unfortunate to move to a segmented memory model. Could this be re-imagined in a linear memory space?

The data segments could be removed from the MVP, and replaced by an interface to access data blobs in the wasm file, and this would appear sufficient to allow the wasm code to load these at dynamic locations at runtime.

The core issue seems to be that it would be useful to allow constant address references to be baked into the compiled code? For example, for a module to be able to reference data loaded in its own data segments at addresses known at compile time, or to be able to bake in the index of variables in the linear memory.

It would appear that this can be solved entirely by rewriting the wasm as it is loaded, and thus this segmented memory model it not necessary for the MVP?

For example, the code could use separate opcodes (macros) to encode linear memory accesses relative to a constant base and the loader would rewrite these adding the base that has been chosen at runtime by the loader.

This would be rather unfortunate for sharing compiled modules, and for caching of compiled modules. If the code is to be shared then these can not be baked into the code anyway, which just leaves the use case of caching which might allow a copy and patch rather than a re-compile which the `get_const` idea seems to support.

So the segmented memory model seems entirely unnecessary for a MVP with dynamic loading, for both the use cases of: top runtime performance by rewriting the segment bases into the offsets; or for the use case of sharing code in which case the offsets are dynamic anyway.